### PR TITLE
cease cloning the ExtendedTruffleContract

### PR DESCRIPTION
**Pull request by dkent600**
Addresses: #98 

Also:
- applies a modified tslint rule
- adds a `vote` method to `AbsoluteVote`

### CODE REVIEW

File: test/helpers.js [hunk 1]
```diff
@@ -111,22 +111,29 @@ export async function getSchemeVotingMachine(dao, scheme, ndxVotingMachineParame
...
+  } else {
+    return await votingMachine.vote(proposalId, vote, { from: voter });
+  }
 }
 
 export async function voteWasExecuted(votingMachine, proposalId) {
   return new Promise(async (resolve) => {
-    const event = votingMachine.ExecuteProposal({ "_proposalId": proposalId }, { fromBlock: 0 });
+    let event;
+    if (votingMachine.contract) {
```
Comment (tibetsprague) [inline test/helpers.js]: I dont really understand this conditional, can you comment?

File: test/helpers.js [hunk 2]
```diff
@@ -111,22 +111,29 @@ export async function getSchemeVotingMachine(dao, scheme, ndxVotingMachineParame
...
 
 /**
  * vote for the proposal given by proposalId.
- * votingMachine must be the raw contract, not a wrapper.
  */
 export async function vote(votingMachine, proposalId, vote, voter) {
   voter = (voter ? voter : accounts[0]);
-  votingMachine = votingMachine.contract ? votingMachine.contract : votingMachine;
-  return await votingMachine.vote(proposalId, vote, { from: voter });
+  if (votingMachine.contract) {
```
Comment (tibetsprague) [inline test/helpers.js]: I dot really understand this conditional, can you comment?

Comment (dkent600) [reply test/helpers.js]: added a comment in the code

File: test/helpers.js (hunk 1 above)
Comment (dkent600) [reply test/helpers.js]: added a comment in the code

**Pull request merged**